YesterdayApi/Core/Identity/TokenDecoder.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using YesterdayApi.Utilities.AutomaticDI;
+using YesterdayApi.Utilities.Exceptions;
+
+namespace YesterdayApi.Core.Identity
+{
+    public class TokenDecoder : ITokenDecoder, ITransient
+    {
+        public int GetUserId(string authHeader)
+        {
+            if (string.IsNullOrEmpty(authHeader))
+            {
+                throw new UnauthorizedException("Error", "Authorization header is empty.");
+            }
+
+            var decodedToken = GetDecodedToken(authHeader);
+
+            if (TokenTooOld(decodedToken.ValidTo))
```

Comment:
Maybe method IsTokenTooOld or IsTokenExpired?

Reply:
I check if the token expired less than 5 days ago, because i want to keep the access token lifetime quite short, so it's different things.

Reply:
Can we instead of this shorten lifetime of token (i mean expired date)?

Reply:
Can you rephrase please, I don't get what you mean.

Reply:
Ahhh... I've got idea right now haha. Nvm. It's fine